### DISCUSSION TITLE
HTML "base" header tag support

### DISCUSSION THREAD

**Discussion by AmebaBrain** [mkdocs/mkdocs#3723] [Q&A] [answered]
Is there a setting which allows to configure `<base href="/">` header tag setup? Existing `site_url` adds just `link rel="canonical"` to the header. But it doesn't help in resolving by browser references to the CSS files included.

If your site is for example `https://mysite.com/blog` and in the browser url you don't have trailing slash `/` your css will not be resolved correctly. Only if you explicitly enter in the browser url `https://mysite.com/blog/`
OR
On your web server configuration you setup 301 redirect from `https://mysite.com/blog` to `https://mysite.com/blog/`. But this implementation will cause browser to request your page twice.

With `<base href="/blog" >` setting in place, even with `https://mysite.com/blog` url in a browser, your css files will be resolved correcty against `/blog` folder on your server and not a root one.

**Comment by Andre601** [comment, marked as answer]
I think the easiest aproach here is to just extend the theme you are using to implement the content yourself.

**Reply by AmebaBrain** [reply]
You are right. I'm using Material for MkDocs. So, I created `overrides/main.html` with such content. It adds `<base href="/blog/">` to the very top of a page, which is required by MDN [specification](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/base). But if you will try to use purpose build `extraheads` block, then it will be added at the end of a head. After CSS and JS files are defined. Still works but violates MDN recommendation.

```
{% extends "base.html" %}

{% block site_meta %}
    <base href="/blog/">
    {{ super() }}
{% endblock %}
```

Despite of the fact that this works, it shows in the console for some reason 3 XHR requests done, instead of expected 1. Looks like `<base href>` acts actually like web server redirect response. So, I kept my web server `/blog` => `/blog/` redirection in place. In such scenario, I have 2 XHR requests and not 3.
